From 4da920d7cc3b7238fa619b7fdfe30b1bbbed9a74 Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Wed, 14 Oct 2015 13:46:51 +0530
Subject: [PATCH] [EM/Charger]: Allow to control cc regardless of charger
 present

When thermal service tries to control(throttle/de-throttle) the charge
current based on the temperature upon charger disconnect is doesn't
effect. This also causes an issue when thermal service throttles the
charger current when charger is connected and if it de-throttle after
charger disconnect, next charger plug charges with the previous
throttled value. To resolve this issue allow thermal service to
control charger current regardless of charger present.

Change-Id: Id97070843ea59e682188b26e00171424ef4f1eb7
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-6770
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
Reviewed-on: https://android.intel.com:443/423756
(cherry picked from commit ef3dfbb3df4a1ea6509aad52898371839bd09d84)
---
 drivers/power/power_supply_charger.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/power/power_supply_charger.c b/drivers/power/power_supply_charger.c
index 2f0cc0e..26a9652 100644
--- a/drivers/power/power_supply_charger.c
+++ b/drivers/power/power_supply_charger.c
@@ -973,8 +973,8 @@ int psy_charger_throttle_charger(struct power_supply *psy,
 {
 	int ret = 0;
 
-	if (!IS_PRESENT(psy))
-		return 0;
+	if (!psy)
+		return -EINVAL;
 
 	if (state < 0 || state >= MAX_THROTTLE_STATE(psy))
 		return -EINVAL;
-- 
1.9.1

